src/main/webapp/app/shared/badge/NotCuratableBadge.tsx

```diff
@@ -1,19 +1,76 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import DefaultBadge from './DefaultBadge';
+import { IRootStore } from 'app/stores';
+import { componentInject } from '../util/typed-inject';
+import { getMutationName } from '../util/firebase/firebase-utils';
+import { RADIO_OPTION_NONE } from 'app/config/constants/constants';
+import { observer } from 'mobx-react';
 
 const NOT_CURATABLE_TOOLTIP_TEXT = {
-  stringMutations: 'Each mutation should have its own mutation effect',
+  stringMutations: 'Each mutation should have its own mutation effect curated',
   default: 'This mutation cannot have its own mutation effect',
 };
 
-export interface INotCuratableBadgeProps {
+export interface INotCuratableBadgeProps extends StoreProps {
   mutationName: string;
 }
 
-const NotCuratableBadge: React.FunctionComponent<INotCuratableBadgeProps> = (props: INotCuratableBadgeProps) => {
-  const { mutationName } = props;
+const NotCuratableBadge: React.FunctionComponent<INotCuratableBadgeProps> = ({ mutationName, mutations }: INotCuratableBadgeProps) => {
   const text = mutationName?.includes(',') ? NOT_CURATABLE_TOOLTIP_TEXT.stringMutations : NOT_CURATABLE_TOOLTIP_TEXT.default;
-  return <DefaultBadge color={'warning'} text={'Not Curatable'} tooltipOverlay={<div>{text}</div>} />;
+
+  const [mutationsWithoutEffect, mutationsWithEffect] = useMemo(() => {
+    const mutationStrings: [string[], string[]] = [[], []];
+    for (const name of mutationName.split(',')) {
+      const trimmedName = name.trim();
+      const foundMutation = mutations?.find(mut => getMutationName(mut.name, mut.alterations) === trimmedName);
+      if (!foundMutation) {
+        continue;
+      }
```

Comment:
If one of the mutation is not found(DNE in our mutation list), then it should be part of the `Need curation` list as well.